cpp/src/arrow/compute/kernels/vector_selection_take_internal.cc

```diff
@@ -884,13 +756,11 @@ void PopulateTakeKernels(std::vector<SelectionKernelData>* out) {
   auto take_indices = match::Integer();
 
   *out = {
-      {InputType(match::Primitive()), take_indices, PrimitiveTakeExec},
+      {InputType(match::Primitive()), take_indices, FixedWidthTakeExec},
       {InputType(match::BinaryLike()), take_indices, VarBinaryTakeExec},
       {InputType(match::LargeBinaryLike()), take_indices, LargeVarBinaryTakeExec},
-      {InputType(Type::FIXED_SIZE_BINARY), take_indices, FSBTakeExec},
+      {InputType(match::FixedSizeBinaryLike()), take_indices, FixedWidthTakeExec},
       {InputType(null()), take_indices, NullTakeExec},
-      {InputType(Type::DECIMAL128), take_indices, PrimitiveTakeExec},
-      {InputType(Type::DECIMAL256), take_indices, PrimitiveTakeExec},
```

Comment:
"FixedSizeBinaryLike" includes FSB and the Decimal types. Not to be confused with "fixed-WIDTH like" which includes special fixed-size lists (see `util/fixed_width_internal.h`). I wanted to name the latter "fixed-width modulo nesting" to be more upfront about the nesting aspect, but that turned out too obtuse and verbose.

Reply:
You could perhaps call it "generalized fixed-width" or something (not for this PR though :-)).

Reply:
Taking note 👍

Reply:
Created an issue #41963